server/services/store/sqlstore/data_migrations.go

```diff
@@ -514,10 +514,15 @@ func (s *SQLStore) getBestTeamForBoard(tx sq.BaseRunner, board *model.Board) (st
 
 func (s *SQLStore) getBoardUserTeams(tx sq.BaseRunner, board *model.Board) (map[string][]string, error) {
 	query := s.getQueryBuilder(tx).
-		Select("TeamMembers.UserId", "TeamMembers.TeamId").
-		From("ChannelMembers").
-		Join("TeamMembers ON ChannelMembers.UserId = TeamMembers.UserId").
-		Where(sq.Eq{"ChannelId": board.ChannelID})
+		Select("tm.UserId", "tm.TeamId").
```

Comment:
Would it make sense to also modify `getBestTeamForBoard` line 490 so that instead of just returning the first team left after filtering, it selects the team based on highest number of overall users.  In other words, a very small team is more likely to be deleted versus a large one.  On that note, what happens if/when the team selected is deleted later?  Do we move the board to some arbitrary team like we do at migration time?  Should we ask the user to pick?

Reply:
On the first question, I agree on a more populated board being a better guess than just the first, although at this point in time I'm a bit reluctant to make more changes than the strictly required. As per the second question, once the migration has finished, the board is just a private board part of the team that it belongs to, with no special considerations.

I think the best solution for both cases is to provide the user with the ability to move boards between teams, so independently of the decision that this migration will take on a specific board, the user will have the final say on where the board should be.